SmartDeviceLinkTests/SDLSubscribeButtonManagerSpec.m

```diff
@@ -0,0 +1,716 @@
+
+#import "SDLError.h"
+#import "SDLOnButtonEvent.h"
+#import "SDLOnButtonPress.h"
+#import "SDLRPCNotificationNotification.h"
+#import "SDLSubscribeButtonManager.h"
+#import "SDLSubscribeButtonObserver.h"
+#import "SDLSubscribeButtonResponse.h"
+#import "SDLSubscribebutton.h"
+#import "SDLUnsubscribeButton.h"
+#import "SDLUnsubscribeButtonResponse.h"
+#import "TestConnectionManager.h"
+#import "TestSubscribeButtonObserver.h"
+
+@interface SDLSubscribeButtonManager()
+
+@property (weak, nonatomic) id<SDLConnectionManagerType> connectionManager;
+@property (strong, nonatomic) NSMutableDictionary<SDLButtonName, NSMutableArray<SDLSubscribeButtonObserver *> *> *subscribeButtonObservers;
+
+@end
+
+QuickSpecBegin(SDLSubscribeButtonManagerSpec)
+
+describe(@"subscribe button manager", ^{
+    __block SDLSubscribeButtonManager *testManager = nil;
+    __block TestConnectionManager *testConnectionManager = nil;
+
+    __block SDLSubscribeButtonResponse *testSubscribeSuccessResponse = nil;
+    __block SDLSubscribeButtonResponse *testSubscribeFailureResponse = nil;
+    __block SDLUnsubscribeButtonResponse *testUnSubscribeSuccessResponse = nil;
+    __block SDLUnsubscribeButtonResponse *testUnSubscribeFailureResponse = nil;
+
+    beforeEach(^{
+        testSubscribeSuccessResponse = [[SDLSubscribeButtonResponse alloc] init];
+        testSubscribeSuccessResponse.success = @YES;
+        testUnSubscribeSuccessResponse.resultCode = SDLResultSuccess;
+
+        testSubscribeFailureResponse = [[SDLSubscribeButtonResponse alloc] init];
+        testSubscribeFailureResponse.success = @NO;
+        testUnSubscribeSuccessResponse.resultCode = SDLResultDisallowed;
+
+        testUnSubscribeSuccessResponse = [[SDLUnsubscribeButtonResponse alloc] init];
+        testUnSubscribeSuccessResponse.success = @YES;
+        testUnSubscribeSuccessResponse.resultCode = SDLResultSuccess;
+
+        testUnSubscribeFailureResponse = [[SDLUnsubscribeButtonResponse alloc] init];
+        testUnSubscribeFailureResponse.success = @NO;
+        testUnSubscribeFailureResponse.resultCode = SDLResultDisallowed;
+    });
+
+    context(@"lifecycle", ^{
+        beforeEach(^{
+            testConnectionManager = [[TestConnectionManager alloc] init];
+            testManager = [[SDLSubscribeButtonManager alloc] initWithConnectionManager:testConnectionManager];
+        });
+
+        it(@"should instantiate correctly with initWithConnectionManager:", ^{
+            expect(testManager.connectionManager).to(equal(testConnectionManager));
+            expect(testManager.subscribeButtonObservers).toNot(beNil());
+        });
+
+        it(@"should do nothing when Start is called", ^{
+            [testManager start];
+        });
+
+        it(@"should clear the list of observers when Stop is called", ^{
+            [testManager stop];
+            expect(testManager.subscribeButtonObservers.count).to(equal(0));
+        });
+    });
+
+    context(@"subscriptions", ^{
```

Comment:
This context isn't helpful because every remaining test is within it. Since the above context isn't useful, this context can go away as well.

Reply:
Removed